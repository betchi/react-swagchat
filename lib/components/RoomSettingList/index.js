"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var RoomSettingList_1 = require("./RoomSettingList");
exports.RoomSettingList = RoomSettingList_1.RoomSettingList;
var RoomSettingModalItem_1 = require("./RoomSettingModalItem");
exports.RoomSettingModalItem = RoomSettingModalItem_1.RoomSettingModalItem;
//# sourceMappingURL=index.js.map