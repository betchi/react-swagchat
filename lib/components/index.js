"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Avatar_1 = require("./Avatar/Avatar");
exports.Avatar = Avatar_1.Avatar;
var Badge_1 = require("./Badge/Badge");
exports.Badge = Badge_1.Badge;
var Button_1 = require("./Button/Button");
exports.Button = Button_1.Button;
var CheckListItem_1 = require("./CheckListItem/CheckListItem");
exports.CheckListItem = CheckListItem_1.CheckListItem;
var ContactList_1 = require("./ContactList/ContactList");
exports.ContactList = ContactList_1.ContactList;
var AddCircle_1 = require("./icons/AddCircle");
exports.AddCircle = AddCircle_1.AddCircle;
var Back_1 = require("./icons/Back");
exports.Back = Back_1.Back;
var Block_1 = require("./icons/Block");
exports.Block = Block_1.Block;
var Camera_1 = require("./icons/Camera");
exports.Camera = Camera_1.Camera;
var CheckCircle_1 = require("./icons/CheckCircle");
exports.CheckCircle = CheckCircle_1.CheckCircle;
var CheckCircleOutline_1 = require("./icons/CheckCircleOutline");
exports.CheckCircleOutline = CheckCircleOutline_1.CheckCircleOutline;
var Close_1 = require("./icons/Close");
exports.Close = Close_1.Close;
var Done_1 = require("./icons/Done");
exports.Done = Done_1.Done;
var Edit_1 = require("./icons/Edit");
exports.Edit = Edit_1.Edit;
var Exit_1 = require("./icons/Exit");
exports.Exit = Exit_1.Exit;
var Expand_1 = require("./icons/Expand");
exports.Expand = Expand_1.Expand;
var Keyboard_1 = require("./icons/Keyboard");
exports.Keyboard = Keyboard_1.Keyboard;
var Photo_1 = require("./icons/Photo");
exports.Photo = Photo_1.Photo;
var RadioButtonChecked_1 = require("./icons/RadioButtonChecked");
exports.RadioButtonChecked = RadioButtonChecked_1.RadioButtonChecked;
var RadioButtonUnChecked_1 = require("./icons/RadioButtonUnChecked");
exports.RadioButtonUnChecked = RadioButtonUnChecked_1.RadioButtonUnChecked;
var Send_1 = require("./icons/Send");
exports.Send = Send_1.Send;
var MessageBody_1 = require("./Message/MessageBody");
exports.MessageBody = MessageBody_1.MessageBody;
var MessageDateSeparator_1 = require("./Message/MessageDateSeparator");
exports.MessageDateSeparator = MessageDateSeparator_1.MessageDateSeparator;
var MessageInteractionBottom_1 = require("./Message/MessageInteractionBottom");
exports.MessageInteractionBottom = MessageInteractionBottom_1.MessageInteractionBottom;
var MessageInteractionTop_1 = require("./Message/MessageInteractionTop");
exports.MessageInteractionTop = MessageInteractionTop_1.MessageInteractionTop;
var MessageItem_1 = require("./Message/MessageItem");
exports.MessageItem = MessageItem_1.MessageItem;
var MessageMenuBottom_1 = require("./Message/MessageMenuBottom");
exports.MessageMenuBottom = MessageMenuBottom_1.MessageMenuBottom;
var MessageMenuTop_1 = require("./Message/MessageMenuTop");
exports.MessageMenuTop = MessageMenuTop_1.MessageMenuTop;
var ModalDialog_1 = require("./Modal/ModalDialog");
exports.ModalDialog = ModalDialog_1.ModalDialog;
var ModalView_1 = require("./Modal/ModalView");
exports.ModalView = ModalView_1.ModalView;
var PhotoEdit_1 = require("./PhotoEdit/PhotoEdit");
exports.PhotoEdit = PhotoEdit_1.PhotoEdit;
var RoomEdit_1 = require("./RoomEdit/RoomEdit");
exports.RoomEdit = RoomEdit_1.RoomEdit;
var RoomItem_1 = require("./RoomList/RoomItem");
exports.RoomItem = RoomItem_1.RoomItem;
var RoomList_1 = require("./RoomList/RoomList");
exports.RoomList = RoomList_1.RoomList;
var RoomSettingList_1 = require("./RoomSettingList/RoomSettingList");
exports.RoomSettingList = RoomSettingList_1.RoomSettingList;
var IconListItem_1 = require("./SimpleListItem/IconListItem");
exports.IconListItem = IconListItem_1.IconListItem;
var SimpleListItem_1 = require("./SimpleListItem/SimpleListItem");
exports.SimpleListItem = SimpleListItem_1.SimpleListItem;
var SubTitleBar_1 = require("./SubTitleBar/SubTitleBar");
exports.SubTitleBar = SubTitleBar_1.SubTitleBar;
var TopBar_1 = require("./TopBar/TopBar");
exports.TopBar = TopBar_1.TopBar;
//# sourceMappingURL=index.js.map