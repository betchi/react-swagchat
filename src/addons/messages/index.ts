export { PluginMessageImage, ImageMenu, ImageInteraction, ImageItem } from './Image';
export { PluginMessageSpeech, SpeechMenu, SpeechInteraction, SpeechItem } from './Speech';
export { PluginMessageText, TextMenu, TextInteraction, TextItem } from './Text';

export interface IPluginMessageParams {
  position?: 'top' | 'bottom';
  isAlwaysDisplay?: boolean;
}