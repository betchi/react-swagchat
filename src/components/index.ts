export { Avatar } from './Avatar/Avatar';
export { Badge } from './Badge/Badge';
export { Button } from './Button/Button';
export { ContactList, IContactListProps } from './ContactList/ContactList';
export { MessageBody } from './Message/MessageBody';
export { MessageInteraction } from './Message/internal/MessageInteraction';
export { MessageItem } from './Message/internal/MessageItem';
export { MessageMenu } from './Message/internal/MessageMenu';
export { Modal } from './Modal/Modal';
export { ModalAction, IModalAction } from './Modal/ModalAction';
export { PhotoEdit } from './PhotoEdit/PhotoEdit';
export { RoomList, IRoomListProps } from './RoomList/RoomList';
export { RoomSettingButtons, IRoomSettingButtonsProps } from './RoomSettingButtons/RoomSettingButtons';
export { SubTitleBar } from './SubTitleBar/SubTitleBar';
export { TextAvatar } from './TextAvatar/TextAvatar';
export { TopBar } from './TopBar/TopBar';

export interface IOnClickProps {
  onClick?: (param?: any) => void;
}

export interface IRootStyleProps {
  className?: string;
  style?: Object;
}