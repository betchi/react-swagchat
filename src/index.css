@import './pages/**/*.css';
@import './components/**/*.css';
@import './containers/**/*.css';
@import './plugins/**/**/*.css';

body {
  font-family: 'Slabo 13px', 'Rounded Mplus 1c', serif;
}
.nodata-wrap {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 500px;
}
.nodata-text {
  opacity: 0.4;
}
.nodata-image {
  width: 80%;
  opacity: 0.3;
  position: fixed;
}
:placeholder-shown,
::-webkit-input-placeholder {
  color: #cccccc;
}

.layout-box-1 {
  padding: 10px 0;
  background-color: #fafafa;
}

.page-container {
  min-width: 320px;
}

.button {
  border: 10px;
  box-sizing: border-box;
  display: inline-block;
  font-family: Roboto, sans-serif;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  cursor: pointer;
  text-decoration: none;
  padding: 5px;
  outline: none;
  font-size: 0px;
  font-weight: inherit;
  position: relative;
  z-index: 1;
  overflow: visible;
  transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
  width: 38px;
  height: 38px;
  background: none;
}
.icon {
  display: inline-block;
  color: #0084ff;
  fill: currentcolor;
  user-select: none;
  transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
}
.button_text {
  color: #0084ff;
  font-size: 24px;
  width: 50px;
  height: 38px;
}