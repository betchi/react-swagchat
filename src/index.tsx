import {
  renderTemplateMessenger,
  renderTemplateSimpleMessage,
} from './templates/';
import {
  Avatar,
  Badge,
  Button,
  ContactList,
  MessageBody,
  Modal,
  ModalAction,
  PhotoEdit,
  RoomEditForm,
  RoomList,
  RoomSettingButtons,
  SubTitleBar,
  TextAvatar,
  TopBar,
} from './components';
import {
  PluginMessageText,
  PluginMessageImage,
  TextItem,
  ImageItem,
} from './plugins/message';
import {
  PluginRoomListItemRoomAndUserNameWithMessage,
  PluginRoomListItemRoomNameWithMessage,
} from './plugins/roomListItem';

export * from './components';
export * from './templates';

// For browser
(window as any).Swag = {};

(window as any).Swag.renderTemplateMessenger = renderTemplateMessenger;
(window as any).Swag.renderTemplateSimpleMessage = renderTemplateSimpleMessage;

(window as any).Swag.Avatar = Avatar;
(window as any).Swag.Badge = Badge;
(window as any).Swag.Button = Button;
(window as any).Swag.ContactList = ContactList;
(window as any).Swag.MessageBody = MessageBody;
(window as any).Swag.Modal = Modal;
(window as any).Swag.ModalAction = ModalAction;
(window as any).Swag.PhotoEdit = PhotoEdit;
(window as any).Swag.RoomEditForm = RoomEditForm;
(window as any).Swag.RoomList = RoomList;
(window as any).Swag.RoomSettingButtons = RoomSettingButtons;
(window as any).Swag.SubTitleBar = SubTitleBar;
(window as any).Swag.TextAvatar = TextAvatar;
(window as any).Swag.TopBar = TopBar;

(window as any).Swag.PluginRoomListItemRoomAndUserNameWithMessage = PluginRoomListItemRoomAndUserNameWithMessage;
(window as any).Swag.PluginRoomListItemRoomNameWithMessage = PluginRoomListItemRoomNameWithMessage;
(window as any).Swag.PluginMessageText = PluginMessageText;
(window as any).Swag.PluginMessageImage = PluginMessageImage;
(window as any).Swag.TextItem = TextItem;
(window as any).Swag.ImageItem = ImageItem;
