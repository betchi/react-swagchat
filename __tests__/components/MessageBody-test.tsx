declare var jest, describe, it, expect, require;

import * as React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import { MessageBody } from '../../src/components';
import { IMessage, IUserForRoom } from 'swagchat-sdk';
import { PluginMessageText, PluginMessageImage } from '../../src/addons/messages/';

const roomUsers: {[key: string]: IUserForRoom} = {
  ['00581ea9-3547-4c81-930c-a3ed042e4b21']: {
    userId: '00581ea9-3547-4c81-930c-a3ed042e4b21',
    name: 'Williams',
    pictureUrl: 'http://example.com/williams.png'
  },
  ['7da9e9c6-d174-4ab5-86f8-99e727c710ff']: {
    userId: '7da9e9c6-d174-4ab5-86f8-99e727c710ff',
    name: 'Lee',
    pictureUrl: 'http://example.com/lee.png'
  },
};

const messages: {[key: string]: IMessage} = {
  ['61952241-48c8-4cdd-9fe2-ab61b8bf9c55']: {
    messageId: '61952241-48c8-4cdd-9fe2-ab61b8bf9c55',
    roomId: '821d92d7-4a68-452a-b660-51379544cec4',
    userId: '00581ea9-3547-4c81-930c-a3ed042e4b21',
    type:  'text',
    payload: {
      text: 'How\'ve you been recently?'
    },
    created: '2017-06-30T11:39:31Z'
  },
  ['9d79f275-a80c-41ae-aa26-0631d94486b4']: {
    messageId: '9d79f275-a80c-41ae-aa26-0631d94486b4',
    roomId: '821d92d7-4a68-452a-b660-51379544cec4',
    userId: '7da9e9c6-d174-4ab5-86f8-99e727c710ff',
    type: 'text',
    payload: {
      text: 'I\'ve been good. And you?'
    },
    created: '2017-06-30T11:42:09Z'
  },
  ['d26c3d0e-7568-49be-ada7-e9e2ef023cda']: {
    messageId: 'd26c3d0e-7568-49be-ada7-e9e2ef023cda',
    roomId: '821d92d7-4a68-452a-b660-51379544cec4',
    userId: '00581ea9-3547-4c81-930c-a3ed042e4b21',
    type: 'text',
    payload: {
      text: 'I\'ve been working all day.'
    },
    created: '2017-06-30T11:46:15Z'
  },
  ['7ecc5c82-8861-4161-9d18-98dc4f160976']: {
    messageId: '7ecc5c82-8861-4161-9d18-98dc4f160976',
    roomId: '821d92d7-4a68-452a-b660-51379544cec4',
    userId: '7da9e9c6-d174-4ab5-86f8-99e727c710ff',
    type: 'text',
    payload: {
      text: 'Don\'t push yourself too hard. You should get some rest.'
    },
    created: '2017-06-30T11:46:44Z'
  },
  ['00dfce2d-2635-4adf-b999-3e0f0bddabb6']: {
    messageId: '00dfce2d-2635-4adf-b999-3e0f0bddabb6',
    roomId: '821d92d7-4a68-452a-b660-51379544cec4',
    userId: '7da9e9c6-d174-4ab5-86f8-99e727c710ff',
    type: 'text',
    payload: {
      text: 'By the way, recently I started keeping dogs.'
    },
    created: '2017-06-30T12:05:56Z'
  },
  ['21af4ffd-71c7-4c3d-bf06-511f6da00100']: {
    messageId: 'fd10d8f2-1924-4398-bac0-d941b4c6f512',
    roomId: '821d92d7-4a68-452a-b660-51379544cec4',
    userId: '00581ea9-3547-4c81-930c-a3ed042e4b21',
    type: 'text',
    payload: {
      text: 'That\'s nice.'
    },
    created: '2017-08-15T08:19:26Z'
  },
};

const testComponent1 = shallow(
  <MessageBody
    myUserId="myUserId"
    roomUsers={roomUsers}
    messages={messages}
    pluginMessages={[new PluginMessageText, new PluginMessageImage]}
  />
);
it('MessageBody renders correctly', () => {
  const j = toJson(testComponent1);
  expect(j).toMatchSnapshot();
});

const testComponent2 = shallow(
  <MessageBody
    myUserId="myUserId"
    roomUsers={roomUsers}
    messages={messages}
    pluginMessages={[new PluginMessageText, new PluginMessageImage]}
    customPluginMessages={[new PluginMessageText, new PluginMessageImage]}
    noMessageText="No messages"
    noMessageImage="http://example.com/noimage.png"
    className="test-class"
    style={{width: '100px'}}
  />
);
it('MessageBody renders correctly', () => {
  const j = toJson(testComponent2);
  expect(j).toMatchSnapshot();
});
