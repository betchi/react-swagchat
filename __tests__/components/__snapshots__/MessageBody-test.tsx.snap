// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBody renders correctly 1`] = `
<div
  className=""
  style={Object {}}
>
  <MessageDateSeparator
    className=""
    date="6/30"
    key="date-separator-61952241-48c8-4cdd-9fe2-ab61b8bf9c55"
    style={Object {}}
  />
  <MessageItem
    key="message-item-61952241-48c8-4cdd-9fe2-ab61b8bf9c55"
    message={
      Object {
        "created": "2017-06-30T11:39:31Z",
        "messageId": "61952241-48c8-4cdd-9fe2-ab61b8bf9c55",
        "payload": Object {
          "text": "How've you been recently?",
        },
        "roomId": "821d92d7-4a68-452a-b660-51379544cec4",
        "type": "text",
        "userId": "00581ea9-3547-4c81-930c-a3ed042e4b21",
      }
    }
    myUserId="myUserId"
    pluginMessages={
      Array [
        PluginMessageText {
          "interaction": [Function],
          "isAlwaysDisplay": true,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "text",
          "position": "bottom",
        },
        PluginMessageImage {
          "interaction": [Function],
          "isAlwaysDisplay": false,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "image",
          "position": "bottom",
        },
      ]
    }
    user={
      Object {
        "name": "Williams",
        "pictureUrl": "http://example.com/williams.png",
        "userId": "00581ea9-3547-4c81-930c-a3ed042e4b21",
      }
    }
  />
  <MessageItem
    key="message-item-9d79f275-a80c-41ae-aa26-0631d94486b4"
    message={
      Object {
        "created": "2017-06-30T11:42:09Z",
        "messageId": "9d79f275-a80c-41ae-aa26-0631d94486b4",
        "payload": Object {
          "text": "I've been good. And you?",
        },
        "roomId": "821d92d7-4a68-452a-b660-51379544cec4",
        "type": "text",
        "userId": "7da9e9c6-d174-4ab5-86f8-99e727c710ff",
      }
    }
    myUserId="myUserId"
    pluginMessages={
      Array [
        PluginMessageText {
          "interaction": [Function],
          "isAlwaysDisplay": true,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "text",
          "position": "bottom",
        },
        PluginMessageImage {
          "interaction": [Function],
          "isAlwaysDisplay": false,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "image",
          "position": "bottom",
        },
      ]
    }
    user={
      Object {
        "name": "Lee",
        "pictureUrl": "http://example.com/lee.png",
        "userId": "7da9e9c6-d174-4ab5-86f8-99e727c710ff",
      }
    }
  />
  <MessageItem
    key="message-item-d26c3d0e-7568-49be-ada7-e9e2ef023cda"
    message={
      Object {
        "created": "2017-06-30T11:46:15Z",
        "messageId": "d26c3d0e-7568-49be-ada7-e9e2ef023cda",
        "payload": Object {
          "text": "I've been working all day.",
        },
        "roomId": "821d92d7-4a68-452a-b660-51379544cec4",
        "type": "text",
        "userId": "00581ea9-3547-4c81-930c-a3ed042e4b21",
      }
    }
    myUserId="myUserId"
    pluginMessages={
      Array [
        PluginMessageText {
          "interaction": [Function],
          "isAlwaysDisplay": true,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "text",
          "position": "bottom",
        },
        PluginMessageImage {
          "interaction": [Function],
          "isAlwaysDisplay": false,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "image",
          "position": "bottom",
        },
      ]
    }
    user={
      Object {
        "name": "Williams",
        "pictureUrl": "http://example.com/williams.png",
        "userId": "00581ea9-3547-4c81-930c-a3ed042e4b21",
      }
    }
  />
  <MessageItem
    key="message-item-7ecc5c82-8861-4161-9d18-98dc4f160976"
    message={
      Object {
        "created": "2017-06-30T11:46:44Z",
        "messageId": "7ecc5c82-8861-4161-9d18-98dc4f160976",
        "payload": Object {
          "text": "Don't push yourself too hard. You should get some rest.",
        },
        "roomId": "821d92d7-4a68-452a-b660-51379544cec4",
        "type": "text",
        "userId": "7da9e9c6-d174-4ab5-86f8-99e727c710ff",
      }
    }
    myUserId="myUserId"
    pluginMessages={
      Array [
        PluginMessageText {
          "interaction": [Function],
          "isAlwaysDisplay": true,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "text",
          "position": "bottom",
        },
        PluginMessageImage {
          "interaction": [Function],
          "isAlwaysDisplay": false,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "image",
          "position": "bottom",
        },
      ]
    }
    user={
      Object {
        "name": "Lee",
        "pictureUrl": "http://example.com/lee.png",
        "userId": "7da9e9c6-d174-4ab5-86f8-99e727c710ff",
      }
    }
  />
  <MessageItem
    key="message-item-00dfce2d-2635-4adf-b999-3e0f0bddabb6"
    message={
      Object {
        "created": "2017-06-30T12:05:56Z",
        "messageId": "00dfce2d-2635-4adf-b999-3e0f0bddabb6",
        "payload": Object {
          "text": "By the way, recently I started keeping dogs.",
        },
        "roomId": "821d92d7-4a68-452a-b660-51379544cec4",
        "type": "text",
        "userId": "7da9e9c6-d174-4ab5-86f8-99e727c710ff",
      }
    }
    myUserId="myUserId"
    pluginMessages={
      Array [
        PluginMessageText {
          "interaction": [Function],
          "isAlwaysDisplay": true,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "text",
          "position": "bottom",
        },
        PluginMessageImage {
          "interaction": [Function],
          "isAlwaysDisplay": false,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "image",
          "position": "bottom",
        },
      ]
    }
    user={
      Object {
        "name": "Lee",
        "pictureUrl": "http://example.com/lee.png",
        "userId": "7da9e9c6-d174-4ab5-86f8-99e727c710ff",
      }
    }
  />
  <MessageDateSeparator
    className=""
    date="8/15"
    key="date-separator-fd10d8f2-1924-4398-bac0-d941b4c6f512"
    style={Object {}}
  />
  <MessageItem
    key="message-item-fd10d8f2-1924-4398-bac0-d941b4c6f512"
    message={
      Object {
        "created": "2017-08-15T08:19:26Z",
        "messageId": "fd10d8f2-1924-4398-bac0-d941b4c6f512",
        "payload": Object {
          "text": "That's nice.",
        },
        "roomId": "821d92d7-4a68-452a-b660-51379544cec4",
        "type": "text",
        "userId": "00581ea9-3547-4c81-930c-a3ed042e4b21",
      }
    }
    myUserId="myUserId"
    pluginMessages={
      Array [
        PluginMessageText {
          "interaction": [Function],
          "isAlwaysDisplay": true,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "text",
          "position": "bottom",
        },
        PluginMessageImage {
          "interaction": [Function],
          "isAlwaysDisplay": false,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "image",
          "position": "bottom",
        },
      ]
    }
    user={
      Object {
        "name": "Williams",
        "pictureUrl": "http://example.com/williams.png",
        "userId": "00581ea9-3547-4c81-930c-a3ed042e4b21",
      }
    }
  />
</div>
`;

exports[`MessageBody renders correctly 2`] = `
<div
  className="test-class"
  style={
    Object {
      "width": "100px",
    }
  }
>
  <MessageDateSeparator
    className=""
    date="6/30"
    key="date-separator-61952241-48c8-4cdd-9fe2-ab61b8bf9c55"
    style={Object {}}
  />
  <MessageItem
    customPluginMessages={
      Array [
        PluginMessageText {
          "interaction": [Function],
          "isAlwaysDisplay": true,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "text",
          "position": "bottom",
        },
        PluginMessageImage {
          "interaction": [Function],
          "isAlwaysDisplay": false,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "image",
          "position": "bottom",
        },
      ]
    }
    key="message-item-61952241-48c8-4cdd-9fe2-ab61b8bf9c55"
    message={
      Object {
        "created": "2017-06-30T11:39:31Z",
        "messageId": "61952241-48c8-4cdd-9fe2-ab61b8bf9c55",
        "payload": Object {
          "text": "How've you been recently?",
        },
        "roomId": "821d92d7-4a68-452a-b660-51379544cec4",
        "type": "text",
        "userId": "00581ea9-3547-4c81-930c-a3ed042e4b21",
      }
    }
    myUserId="myUserId"
    pluginMessages={
      Array [
        PluginMessageText {
          "interaction": [Function],
          "isAlwaysDisplay": true,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "text",
          "position": "bottom",
        },
        PluginMessageImage {
          "interaction": [Function],
          "isAlwaysDisplay": false,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "image",
          "position": "bottom",
        },
      ]
    }
    user={
      Object {
        "name": "Williams",
        "pictureUrl": "http://example.com/williams.png",
        "userId": "00581ea9-3547-4c81-930c-a3ed042e4b21",
      }
    }
  />
  <MessageItem
    customPluginMessages={
      Array [
        PluginMessageText {
          "interaction": [Function],
          "isAlwaysDisplay": true,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "text",
          "position": "bottom",
        },
        PluginMessageImage {
          "interaction": [Function],
          "isAlwaysDisplay": false,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "image",
          "position": "bottom",
        },
      ]
    }
    key="message-item-9d79f275-a80c-41ae-aa26-0631d94486b4"
    message={
      Object {
        "created": "2017-06-30T11:42:09Z",
        "messageId": "9d79f275-a80c-41ae-aa26-0631d94486b4",
        "payload": Object {
          "text": "I've been good. And you?",
        },
        "roomId": "821d92d7-4a68-452a-b660-51379544cec4",
        "type": "text",
        "userId": "7da9e9c6-d174-4ab5-86f8-99e727c710ff",
      }
    }
    myUserId="myUserId"
    pluginMessages={
      Array [
        PluginMessageText {
          "interaction": [Function],
          "isAlwaysDisplay": true,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "text",
          "position": "bottom",
        },
        PluginMessageImage {
          "interaction": [Function],
          "isAlwaysDisplay": false,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "image",
          "position": "bottom",
        },
      ]
    }
    user={
      Object {
        "name": "Lee",
        "pictureUrl": "http://example.com/lee.png",
        "userId": "7da9e9c6-d174-4ab5-86f8-99e727c710ff",
      }
    }
  />
  <MessageItem
    customPluginMessages={
      Array [
        PluginMessageText {
          "interaction": [Function],
          "isAlwaysDisplay": true,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "text",
          "position": "bottom",
        },
        PluginMessageImage {
          "interaction": [Function],
          "isAlwaysDisplay": false,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "image",
          "position": "bottom",
        },
      ]
    }
    key="message-item-d26c3d0e-7568-49be-ada7-e9e2ef023cda"
    message={
      Object {
        "created": "2017-06-30T11:46:15Z",
        "messageId": "d26c3d0e-7568-49be-ada7-e9e2ef023cda",
        "payload": Object {
          "text": "I've been working all day.",
        },
        "roomId": "821d92d7-4a68-452a-b660-51379544cec4",
        "type": "text",
        "userId": "00581ea9-3547-4c81-930c-a3ed042e4b21",
      }
    }
    myUserId="myUserId"
    pluginMessages={
      Array [
        PluginMessageText {
          "interaction": [Function],
          "isAlwaysDisplay": true,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "text",
          "position": "bottom",
        },
        PluginMessageImage {
          "interaction": [Function],
          "isAlwaysDisplay": false,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "image",
          "position": "bottom",
        },
      ]
    }
    user={
      Object {
        "name": "Williams",
        "pictureUrl": "http://example.com/williams.png",
        "userId": "00581ea9-3547-4c81-930c-a3ed042e4b21",
      }
    }
  />
  <MessageItem
    customPluginMessages={
      Array [
        PluginMessageText {
          "interaction": [Function],
          "isAlwaysDisplay": true,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "text",
          "position": "bottom",
        },
        PluginMessageImage {
          "interaction": [Function],
          "isAlwaysDisplay": false,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "image",
          "position": "bottom",
        },
      ]
    }
    key="message-item-7ecc5c82-8861-4161-9d18-98dc4f160976"
    message={
      Object {
        "created": "2017-06-30T11:46:44Z",
        "messageId": "7ecc5c82-8861-4161-9d18-98dc4f160976",
        "payload": Object {
          "text": "Don't push yourself too hard. You should get some rest.",
        },
        "roomId": "821d92d7-4a68-452a-b660-51379544cec4",
        "type": "text",
        "userId": "7da9e9c6-d174-4ab5-86f8-99e727c710ff",
      }
    }
    myUserId="myUserId"
    pluginMessages={
      Array [
        PluginMessageText {
          "interaction": [Function],
          "isAlwaysDisplay": true,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "text",
          "position": "bottom",
        },
        PluginMessageImage {
          "interaction": [Function],
          "isAlwaysDisplay": false,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "image",
          "position": "bottom",
        },
      ]
    }
    user={
      Object {
        "name": "Lee",
        "pictureUrl": "http://example.com/lee.png",
        "userId": "7da9e9c6-d174-4ab5-86f8-99e727c710ff",
      }
    }
  />
  <MessageItem
    customPluginMessages={
      Array [
        PluginMessageText {
          "interaction": [Function],
          "isAlwaysDisplay": true,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "text",
          "position": "bottom",
        },
        PluginMessageImage {
          "interaction": [Function],
          "isAlwaysDisplay": false,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "image",
          "position": "bottom",
        },
      ]
    }
    key="message-item-00dfce2d-2635-4adf-b999-3e0f0bddabb6"
    message={
      Object {
        "created": "2017-06-30T12:05:56Z",
        "messageId": "00dfce2d-2635-4adf-b999-3e0f0bddabb6",
        "payload": Object {
          "text": "By the way, recently I started keeping dogs.",
        },
        "roomId": "821d92d7-4a68-452a-b660-51379544cec4",
        "type": "text",
        "userId": "7da9e9c6-d174-4ab5-86f8-99e727c710ff",
      }
    }
    myUserId="myUserId"
    pluginMessages={
      Array [
        PluginMessageText {
          "interaction": [Function],
          "isAlwaysDisplay": true,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "text",
          "position": "bottom",
        },
        PluginMessageImage {
          "interaction": [Function],
          "isAlwaysDisplay": false,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "image",
          "position": "bottom",
        },
      ]
    }
    user={
      Object {
        "name": "Lee",
        "pictureUrl": "http://example.com/lee.png",
        "userId": "7da9e9c6-d174-4ab5-86f8-99e727c710ff",
      }
    }
  />
  <MessageDateSeparator
    className=""
    date="8/15"
    key="date-separator-fd10d8f2-1924-4398-bac0-d941b4c6f512"
    style={Object {}}
  />
  <MessageItem
    customPluginMessages={
      Array [
        PluginMessageText {
          "interaction": [Function],
          "isAlwaysDisplay": true,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "text",
          "position": "bottom",
        },
        PluginMessageImage {
          "interaction": [Function],
          "isAlwaysDisplay": false,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "image",
          "position": "bottom",
        },
      ]
    }
    key="message-item-fd10d8f2-1924-4398-bac0-d941b4c6f512"
    message={
      Object {
        "created": "2017-08-15T08:19:26Z",
        "messageId": "fd10d8f2-1924-4398-bac0-d941b4c6f512",
        "payload": Object {
          "text": "That's nice.",
        },
        "roomId": "821d92d7-4a68-452a-b660-51379544cec4",
        "type": "text",
        "userId": "00581ea9-3547-4c81-930c-a3ed042e4b21",
      }
    }
    myUserId="myUserId"
    pluginMessages={
      Array [
        PluginMessageText {
          "interaction": [Function],
          "isAlwaysDisplay": true,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "text",
          "position": "bottom",
        },
        PluginMessageImage {
          "interaction": [Function],
          "isAlwaysDisplay": false,
          "item": [Function],
          "menu": [Function],
          "messageListMarginBottom": 88,
          "name": "image",
          "position": "bottom",
        },
      ]
    }
    user={
      Object {
        "name": "Williams",
        "pictureUrl": "http://example.com/williams.png",
        "userId": "00581ea9-3547-4c81-930c-a3ed042e4b21",
      }
    }
  />
</div>
`;
